aten/src/ATen/core/jit_type.h

```diff
@@ -259,9 +259,19 @@ struct CAFFE2_API OptionalType: public SingleElementType<TypeKind::OptionalType,
     return create(contained_types[0]);
   }
 
+  bool isSubtypeOf(const TypePtr rhs) const override {
+    if (Type::isSubtypeOf(rhs)) {
+      return true;
+    }
+    if(auto rhs_ = rhs->cast<OptionalType>()) {
+      return getElementType()->isSubtypeOf(rhs_->getElementType());
+    }
+    return false;
+  }
+
   bool operator==(const Type& rhs) const override {
     if (auto opt_rhs = rhs.cast<OptionalType>()) {
-      return getElementType()->isSubtypeOf(opt_rhs->getElementType());
+      return *getElementType() == *opt_rhs->getElementType();
     }
```

Comment:
these are the new changes added to (hopefully) fix the tests that failed due to #22104

Reply:
https://github.com/pytorch/pytorch/pull/22186 removed `==` operator overload since the `OptionalType`'s parent type `SingleElementType` already contains the correct overload [here](https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/core/jit_type.h#L209). you could rebase on after that get merged and it should solve your problem.